service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -362,24 +365,70 @@ public void createReferenceResource(ReferencedResource resource)
     storeResource(resource);
   }
 
+  @WriteTransaction
+  private boolean updateResource(
+      UUID workspaceId, UUID resourceId, Map<String, String> updateParams) {
+
+    var params = new MapSqlParameterSource();
+
+    updateParams.forEach(
+        (k, v) -> {
+          if (!StringUtils.isEmpty(k) && !StringUtils.isEmpty(v)) {
+            params.addValue(k, v);
+          }
+        });
+
+    return updateResourceColumns(workspaceId, resourceId, params);
```

Comment:
Seems like you can collapse `updateResourceColumns` and this method into a single method that populates the params and does the update. Looks like this is the only caller.

In fact, you could also pull `getUpdateParams` into this method, making a single worker method for the two forms of `updateResource`.